### DISCUSSION TITLE
How to set this animation for popupbox in MaterialDesignInXamlToolkit?

### DISCUSSION THREAD

**Discussion by VahidEra** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3218] [Q&A] [answered]
Hi guys,
I have a popupbox in my project and in this project I am using the MaterialDesignInXamlToolkit library and I want it to have this animation when my popupbox is opened. Of course, my guess is that because the outline software is written using the MaterialDesignInXamlToolkit library, asking this question here will help me more. And more importantly, here the specialists and developers of this library will answer the questions and will have more mastery in this matter. Therefore, I apologize in advance if asking this question here may seem unnecessary to expert friends.

![1](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/27216710/78e17317-4f27-42a3-943b-1a5e5b4c4323)

**_I slowed the movie down so its animation is better visible:_**

![2](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/27216710/983250e8-a9fb-4b46-91cc-5dce7bd1858a)

I think two transparency and scale options have been used for this animation, the scale is created step by step and the value has not been suddenly transferred from 0 to 1, of course this is my guess.
I tried to produce this animation, but I did not get the necessary result. The smoothness of the opening of this popupbox is interesting to me, and my animation does not have such smoothness at all Because I only noticed the animation movements by slowing down the opening of the popup box. And I ask you to help me in completing this animation if you can.  For your convenience, I have provided the simple project file below.  Thanks for your help.

[This is Sample Project File](https://drive.google.com/file/d/1uXUdikJP-elN2VCELv_lIsElxgXdFvVT/view?usp=sharing)

**Comment by nicolaihenriksen** [comment, marked as answer]
@VahidEra It is not clear to me exactly what you want to achieve. However, I think the fact that you only give the animation 120ms to complete is probably why it is not "smooth".

Try changing
```xaml
<EasingDoubleKeyFrame KeyTime="0:0:0:0.12" Value="1" />
```

to

```xaml
<EasingDoubleKeyFrame KeyTime="0:0:0:0.35" Value="1" />
```

That should slow it down a bit, but give the animation enough time (i.e. key frames) to produce a visible animation.

![PopupBoxAnimation](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/19572699/e20ad359-f025-4740-88ae-551b1219ae7b)

**Reply by VahidEra** [reply]
@nicolaihenriksen Dear Nicolai,Compare these two animations.
![240191485-e20ad359-f025-4740-88ae-551b1219ae7b](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/27216710/84747aa3-aeb4-4d29-b6b2-34289443a010)  
![AnimationF2](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/27216710/f3288aa6-f317-434a-8561-25ffa2432509)

My animation is very artificial and robot-like.  But the outline animation is very dynamic and fluid. The next point is that I think  also used transparency in the outline animation. I used EasingDoubleKeyFrame but it still gives me a robot-like animation.  And the animation speed is very low.  Can something be done about this?

**Reply by nicolaihenriksen** [reply]
@VahidEra Hmm... I'm no expert on animations, so let's hope someone else can provide some better feedback than me.

**Reply by VahidEra** [reply]
@nicolaihenriksen Dear Nicolai, You are a very humble person.  And you put a lot of effort into improving this library and the last update you provided MaterialDesignInXamlToolkit has improved a lot and I thank you for this.

**Reply by nicolaihenriksen** [reply]
@VahidEra Thank you for the kind words. I enjoy contributing so I'm glad that people find it useful.

**Reply by VahidEra** [reply]
@nicolaihenriksen Dear Nicolai, It is 100% true. You are the best and the best never rest.
